File: ruby/ruby/IceLocal/LocalException.rb
# encoding: utf-8
#
# Copyright (c) ZeroC, Inc.

module Ice
    class InitializationException < LocalException
        def initialize(reason = '')
            @reason = reason
        end

        def to_s
            '::Ice::InitializationException'
        end

        attr_accessor :reason
    end

    class PluginInitializationException < LocalException
        def initialize(reason = '')
            @reason = reason
        end

        def to_s
            '::Ice::PluginInitializationException'
        end

        attr_accessor :reason
    end

    class AlreadyRegisteredException < LocalException
        def initialize(kindOfObject = '', id = '')
            @kindOfObject = kindOfObject
            @id = id
        end

        def to_s
            '::Ice::AlreadyRegisteredException'
        end

        attr_accessor :kindOfObject, :id
    end

    class NotRegisteredException < LocalException
        def initialize(kindOfObject = '', id = '')
            @kindOfObject = kindOfObject
            @id = id
        end

        def to_s
            '::Ice::NotRegisteredException'
        end

        attr_accessor :kindOfObject, :id
    end

    class TwowayOnlyException < LocalException
        def initialize(operation = '')
            @operation = operation
        end

        def to_s
            '::Ice::TwowayOnlyException'
        end

        attr_accessor :operation
    end

    class UnknownException < LocalException
        def initialize(unknown = '')
            @unknown = unknown
        end

        def to_s
            '::Ice::UnknownException'
        end

        attr_accessor :unknown
    end

    class UnknownLocalException < UnknownException
        def initialize(unknown = '')
            super(unknown)
        end

        def to_s
            '::Ice::UnknownLocalException'
        end
    end

    class UnknownUserException < UnknownException
        def initialize(unknown = '')
            super(unknown)
        end

        def to_s
            '::Ice::UnknownUserException'
        end
    end

    class CommunicatorDestroyedException < LocalException
        def initialize
        end

        def to_s
            '::Ice::CommunicatorDestroyedException'
        end
    end

    class ObjectAdapterDeactivatedException < LocalException
        def initialize(name = '')
            @name = name
        end

        def to_s
            '::Ice::ObjectAdapterDeactivatedException'
        end

        attr_accessor :name
    end

    class ObjectAdapterIdInUseException < LocalException
        def initialize(id = '')
            @id = id
        end

        def to_s
            '::Ice::ObjectAdapterIdInUseException'
        end

        attr_accessor :id
    end

    class NoEndpointException < LocalException
        def initialize(proxy = '')
            @proxy = proxy
        end

        def to_s
            '::Ice::NoEndpointException'
        end

        attr_accessor :proxy
    end

    class ParseException < LocalException
        def initialize(str = '')
            @str = str
        end

        def to_s
            '::Ice::ParseException'
        end

        attr_accessor :str
    end

    class IllegalIdentityException < LocalException
        def initialize
        end

        def to_s
            '::Ice::IllegalIdentityException'
        end
    end

    class IllegalServantException < LocalException
        def initialize(reason = '')
            @reason = reason
        end

        def to_s
            '::Ice::IllegalServantException'
        end

        attr_accessor :reason
    end

    class RequestFailedException < LocalException
        def initialize(id, facet, operation)
            @id = id
            @facet = facet
            @operation = operation
        end

        def to_s
            '::Ice::RequestFailedException'
        end

        attr_reader :id, :facet, :operation
    end

    class ObjectNotExistException < RequestFailedException
        def to_s
            '::Ice::ObjectNotExistException'
        end
    end

    class FacetNotExistException < RequestFailedException
        def to_s
            '::Ice::FacetNotExistException'
        end
    end

    class OperationNotExistException < RequestFailedException
        def to_s
            '::Ice::OperationNotExistException'
        end
    end

    class SyscallException < LocalException
        def initialize(error = 0)
            @error = error
        end

        def to_s
            '::Ice::SyscallException'
        end

        attr_accessor :error
    end

    class SocketException < SyscallException
        def initialize(error = 0)
            super(error)
        end

        def to_s
            '::Ice::SocketException'
        end
    end

    class CFNetworkException < SocketException
        def initialize(error = 0, domain = '')
            super(error)
            @domain = domain
        end

        def to_s
            '::Ice::CFNetworkException'
        end

        attr_accessor :domain
    end

    class FileException < SyscallException
        def initialize(error = 0, path = '')
            super(error)
            @path = path
        end

        def to_s
            '::Ice::FileException'
        end

        attr_accessor :path
    end

    class ConnectFailedException < SocketException
        def initialize(error = 0)
            super(error)
        end

        def to_s
            '::Ice::ConnectFailedException'
        end
    end

    class ConnectionRefusedException < ConnectFailedException
        def initialize(error = 0)
            super(error)
        end

        def to_s
            '::Ice::ConnectionRefusedException'
        end
    end

    class ConnectionLostException < SocketException
        def initialize(error = 0)
            super(error)
        end

        def to_s
            '::Ice::ConnectionLostException'
        end
    end

    class DNSException < LocalException
        def initialize(error = 0, host = '')
            @error = error
            @host = host
        end

        def to_s
            '::Ice::DNSException'
        end

        attr_accessor :error, :host
    end

    class ConnectionIdleException < LocalException
        def initialize
        end

        def to_s
            '::Ice::ConnectionIdleException'
        end
    end

    class TimeoutException < LocalException
        def initialize
        end

        def to_s
            '::Ice::TimeoutException'
        end
    end

    class ConnectTimeoutException < TimeoutException
        def initialize
        end

        def to_s
            '::Ice::ConnectTimeoutException'
        end
    end

    class CloseTimeoutException < TimeoutException
        def initialize
        end

        def to_s
            '::Ice::CloseTimeoutException'
        end
    end

    class InvocationTimeoutException < TimeoutException
        def initialize
        end

        def to_s
            '::Ice::InvocationTimeoutException'
        end
    end

    class InvocationCanceledException < LocalException
        def initialize
        end

        def to_s
            '::Ice::InvocationCanceledException'
        end
    end

    class ProtocolException < LocalException
        def initialize(reason = '')
            @reason = reason
        end

        def to_s
            '::Ice::ProtocolException'
        end

        attr_accessor :reason
    end

    class CloseConnectionException < ProtocolException
        def initialize(reason = '')
            super(reason)
        end

        def to_s
            '::Ice::CloseConnectionException'
        end
    end

    class ConnectionManuallyClosedException < LocalException
        def initialize(graceful = false)
            @graceful = graceful
        end

        def to_s
            '::Ice::ConnectionManuallyClosedException'
        end

        attr_accessor :graceful
    end

    class DatagramLimitException < ProtocolException
        def initialize(reason = '')
            super(reason)
        end

        def to_s
            '::Ice::DatagramLimitException'
        end
    end

    class MarshalException < LocalException # TODO: ProtocolException
    end

    class FeatureNotSupportedException < LocalException
        def initialize(unsupportedFeature = '')
            @unsupportedFeature = unsupportedFeature
        end

        def to_s
            '::Ice::FeatureNotSupportedException'
        end

        attr_accessor :unsupportedFeature
    end

    class SecurityException < LocalException
        def initialize(reason = '')
            @reason = reason
        end

        def to_s
            '::Ice::SecurityException'
        end

        attr_accessor :reason
    end

    class FixedProxyException < LocalException
        def initialize
        end

        def to_s
            '::Ice::FixedProxyException'
        end
    end
end

File: php/lib/IceLocal/LocalException.php
<?php
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

namespace
{
    require_once 'Ice/Identity.php';
}

namespace Ice
{
    class RequestFailedException extends LocalException
    {
        public Identity $id;
        public string $facet;
        public string $operation;
    }

    final class ObjectNotExistException extends RequestFailedException
    {
    }

    final class FacetNotExistException extends RequestFailedException
    {
    }

    final class OperationNotExistException extends RequestFailedException
    {
    }

    class InitializationException extends LocalException
    {
        public function __construct($reason='')
        {
            $this->reason = $reason;
        }

        public function ice_id()
        {
            return '::Ice::InitializationException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $reason;
    }

    class PluginInitializationException extends LocalException
    {
        public function __construct($reason='')
        {
            $this->reason = $reason;
        }

        public function ice_id()
        {
            return '::Ice::PluginInitializationException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $reason;
    }

    class AlreadyRegisteredException extends LocalException
    {
        public function __construct($kindOfObject='', $id='')
        {
            $this->kindOfObject = $kindOfObject;
            $this->id = $id;
        }

        public function ice_id()
        {
            return '::Ice::AlreadyRegisteredException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $kindOfObject;
        public $id;
    }

    class NotRegisteredException extends LocalException
    {
        public function __construct($kindOfObject='', $id='')
        {
            $this->kindOfObject = $kindOfObject;
            $this->id = $id;
        }

        public function ice_id()
        {
            return '::Ice::NotRegisteredException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $kindOfObject;
        public $id;
    }

    class TwowayOnlyException extends LocalException
    {
        public function __construct($operation='')
        {
            $this->operation = $operation;
        }

        public function ice_id()
        {
            return '::Ice::TwowayOnlyException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $operation;
    }

    class UnknownException extends LocalException
    {
        public function __construct($unknown='')
        {
            $this->unknown = $unknown;
        }

        public function ice_id()
        {
            return '::Ice::UnknownException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $unknown;
    }

    class UnknownLocalException extends UnknownException
    {
        public function __construct($unknown='')
        {
            parent::__construct($unknown);
        }

        public function ice_id()
        {
            return '::Ice::UnknownLocalException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class UnknownUserException extends UnknownException
    {
        public function __construct($unknown='')
        {
            parent::__construct($unknown);
        }

        public function ice_id()
        {
            return '::Ice::UnknownUserException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class CommunicatorDestroyedException extends LocalException
    {
        public function ice_id()
        {
            return '::Ice::CommunicatorDestroyedException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class ObjectAdapterDeactivatedException extends LocalException
    {
        public function __construct($name='')
        {
            $this->name = $name;
        }

        public function ice_id()
        {
            return '::Ice::ObjectAdapterDeactivatedException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $name;
    }

    class ObjectAdapterIdInUseException extends LocalException
    {
        public function __construct($id='')
        {
            $this->id = $id;
        }

        public function ice_id()
        {
            return '::Ice::ObjectAdapterIdInUseException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $id;
    }

    class NoEndpointException extends LocalException
    {
        public function __construct($proxy='')
        {
            $this->proxy = $proxy;
        }

        public function ice_id()
        {
            return '::Ice::NoEndpointException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $proxy;
    }

    class ParseException extends LocalException
    {
        public function __construct($str='')
        {
            $this->str = $str;
        }

        public function ice_id()
        {
            return '::Ice::ParseException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $str;
    }

    class IllegalIdentityException extends LocalException
    {
        public function __construct($id=null)
        {
        }

        public function ice_id()
        {
            return '::Ice::IllegalIdentityException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class IllegalServantException extends LocalException
    {
        public function __construct($reason='')
        {
            $this->reason = $reason;
        }

        public function ice_id()
        {
            return '::Ice::IllegalServantException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $reason;
    }

    class SyscallException extends LocalException
    {
        public function __construct($error=0)
        {
            $this->error = $error;
        }

        public function ice_id()
        {
            return '::Ice::SyscallException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $error;
    }

    class SocketException extends SyscallException
    {
        public function __construct($error=0)
        {
            parent::__construct($error);
        }

        public function ice_id()
        {
            return '::Ice::SocketException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class FileException extends SyscallException
    {
        public function __construct($error=0, $path='')
        {
            parent::__construct($error);
            $this->path = $path;
        }

        public function ice_id()
        {
            return '::Ice::FileException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $path;
    }

    class ConnectFailedException extends SocketException
    {
        public function __construct($error=0)
        {
            parent::__construct($error);
        }

        public function ice_id()
        {
            return '::Ice::ConnectFailedException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class ConnectionRefusedException extends ConnectFailedException
    {
        public function __construct($error=0)
        {
            parent::__construct($error);
        }

        public function ice_id()
        {
            return '::Ice::ConnectionRefusedException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class ConnectionLostException extends SocketException
    {
        public function __construct($error=0)
        {
            parent::__construct($error);
        }

        public function ice_id()
        {
            return '::Ice::ConnectionLostException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class DNSException extends LocalException
    {
        public function __construct($error=0, $host='')
        {
            $this->error = $error;
            $this->host = $host;
        }

        public function ice_id()
        {
            return '::Ice::DNSException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $error;
        public $host;
    }

    class ConnectionIdleException extends LocalException
    {
        public function ice_id()
        {
            return '::Ice::ConnectionIdleException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class TimeoutException extends LocalException
    {
        public function ice_id()
        {
            return '::Ice::TimeoutException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class ConnectTimeoutException extends TimeoutException
    {
        public function ice_id()
        {
            return '::Ice::ConnectTimeoutException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class CloseTimeoutException extends TimeoutException
    {
        public function ice_id()
        {
            return '::Ice::CloseTimeoutException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class InvocationTimeoutException extends TimeoutException
    {
        public function ice_id()
        {
            return '::Ice::InvocationTimeoutException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class InvocationCanceledException extends LocalException
    {
        public function ice_id()
        {
            return '::Ice::InvocationCanceledException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class ProtocolException extends LocalException
    {
        public function __construct($reason='')
        {
            $this->reason = $reason;
        }

        public function ice_id()
        {
            return '::Ice::ProtocolException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $reason;
    }

    class CloseConnectionException extends ProtocolException
    {
        public function __construct($reason='')
        {
            parent::__construct($reason);
        }

        public function ice_id()
        {
            return '::Ice::CloseConnectionException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class ConnectionManuallyClosedException extends LocalException
    {
        public function __construct($graceful=false)
        {
            $this->graceful = $graceful;
        }

        public function ice_id()
        {
            return '::Ice::ConnectionManuallyClosedException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $graceful;
    }

    class DatagramLimitException extends ProtocolException
    {
        public function __construct($reason='')
        {
            parent::__construct($reason);
        }

        public function ice_id()
        {
            return '::Ice::DatagramLimitException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class MarshalException extends ProtocolException
    {
        public function __construct($reason='')
        {
            parent::__construct($reason);
        }

        public function ice_id()
        {
            return '::Ice::MarshalException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }

    class SecurityException extends LocalException
    {
        public function __construct($reason='')
        {
            $this->reason = $reason;
        }

        public function ice_id()
        {
            return '::Ice::SecurityException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }

        public $reason;
    }

    class FixedProxyException extends LocalException
    {
                public function ice_id()
        {
            return '::Ice::FixedProxyException';
        }

        public function __toString(): string
        {
            return $this->ice_id();
        }
    }
}

?>
